internal/controller/helmchart_controller.go

```diff
@@ -983,15 +986,17 @@ func (r *HelmChartReconciler) namespacedChartRepositoryCallback(ctx context.Cont
 		ctxTimeout, cancel := context.WithTimeout(ctx, obj.Spec.Timeout.Duration)
 		defer cancel()
 
-		clientOpts, err := getter.GetClientOpts(ctxTimeout, r.Client, obj, normalizedURL)
+		clientOpts, certsTmpDir, err := getter.GetClientOpts(ctxTimeout, r.Client, obj, normalizedURL)
 		if err != nil && !errors.Is(err, getter.ErrDeprecatedTLSConfig) {
 			return nil, err
 		}
+		defer os.RemoveAll(certsTmpDir)
```

Comment:
yes, in a multi-tenant environment we should do it.

Reply:
This means changing 

https://github.com/fluxcd/source-controller/blob/38cff765c7419dd90dd947183b6efd037f88c09b/internal/helm/repository/oci_chart_repository.go#L267

So it can clean up all temp files when dealing with dependencies.